bp/src/BPTestBundleConnection.m

```diff
@@ -212,6 +214,51 @@ - (id)_XCT_initializationForUITestingDidFailWithError:(NSError *__strong)errPtr
     return nil;
 }
 
+#pragma mark - Video Recording
+
+static inline NSString* getVideoPath(NSString *directory, NSString *testClass, NSString *method, NSInteger attemptNumber)
+{
+    return [NSString stringWithFormat:@"%@/%@__%@__%ld.mp4", directory, testClass, method, (long)attemptNumber];
+}
+
+- (BOOL)shouldRecordVideo {
+    NSString *videoDirectory = self.context.config.videosDirectory;
+    return videoDirectory != nil && [videoDirectory length] > 0;
+}
+
+- (void)startVideoRecordingForTestClass:(NSString *)testClass method:(NSString *)method
+{
+    [self stopVideoRecording:YES];
+    NSString *videoFileName = getVideoPath(self.context.config.videosDirectory, testClass, method, self.context.attemptNumber);
+    NSString *command = [NSString stringWithFormat:@"xcrun simctl io %@ recordVideo --force --codec=h264 %@", [self.simulator UDID], videoFileName];
+    NSTask *task = [BPUtils buildShellTaskForCommand:command];
+    self.recordVideoTask = task;
+    [task launch];
+    [BPUtils printInfo:INFO withString:@"Started recording video to %@", videoFileName];
+    [BPUtils printInfo:DEBUGINFO withString:@"Started recording video task with command: %@",  [BPUtils getCommandStringForTask:task]];
+}
+
+- (void)stopVideoRecording:(BOOL)forced
+{
+    NSTask *task = self.recordVideoTask;
+    if (task == nil) {
+        if (!forced) {
+            [BPUtils printInfo:ERROR withString: @"Tried to end video recording task normally, but there was no task."];
+        }
+        return;
+    }
+    
+    if (forced) {
+        [BPUtils printInfo:ERROR withString: @"Found dangling video recording task. Stopping it."];
+    }
+    
+    [BPUtils printInfo:INFO withString:@"Stopping recording video."];
+    [BPUtils printInfo:DEBUGINFO withString:@"Stopping video recording task for command: %@", [BPUtils getCommandStringForTask:task]];
```

Comment:
Do we need both the logs?

Reply:
I think the extra info is useful for debug purposes, and DEBUGINFO only shows when it's explicitly enabled.